### ISSUE TITLE
`Field` as {u8, u16, ...} can cause security bugs

### ISSUE THREAD

**Issue by defkit** [noir-lang/noir#6219]
## Problem

Casting variable from `Field` to `u8`, `u16`, `u32` or `u64` should throw an error, if variable doesn't fit.

## Expected Behavior
Error on casting of unfittable variables  (not in 0..255 for `u8` as example)

## Bug
```noir
use dep::std;

fn main(
) {
}

#[test]
fn test_main() {
    assert((0 - 1) as u8 == 0);
    let zero_array: [u8; 32] = [0 as u8; 32];
    let neg_ones_array: [u8; 32] = [(-1) as u8; 32];
    let zero_hash = std::hash::sha256_var(zero_array, 1);
    let neg_ones_hash = std::hash::sha256_var(neg_ones_array, 1);
    assert(zero_hash == neg_ones_hash);

    assert((256) as u8 == 0);
}
```

Or from awesome-noir [example](https://github.com/BattleZips/BattleZips-Noir/blob/99beb27213e61446fa4606711b155b3745620129/circuits/board/src/main.nr#L92) 
`ships` represented as `u4` first, then as `Field`

## To Reproduce
1. Create Noir project for program written in bug section.
2. Run `nargo test`

## Environment
```
nargo version = 0.35.0 
noirc version = 0.35.0+2a0d211b92d002fa75855d4ba27267f8892dd52c (git version hash: 2a0d211b92d002fa75855d4ba27267f8892dd52c, is dirty: false)
```

**Comment by olehmisar** [comment]
Why is it closed? Does casting throw an error now?

**Comment by defkit** [comment]
> Why is it closed? Does casting throw an error now?

No, but https://github.com/noir-lang/noir/pull/8903/files#r2145033595

**Comment by jfecher** [comment]
> Why is it closed? Does casting throw an error now?

It should also be closed because casting should not throw an error. The expected semantics of integer casting in Noir is truncation - not a panic on out of range integers.
